view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/ViewCreatorLauncher.java

```diff
@@ -1,75 +1,49 @@
+import java.lang.reflect.Parameter;
+import java.util.List;
 import org.apache.http.client.HttpClient;
 import org.apache.http.client.methods.HttpPost;
 import org.apache.http.impl.client.HttpClients;
-import org.hypertrace.core.viewcreator.pinot.PinotMultiTableCreationTool;
-import org.hypertrace.core.viewcreator.pinot.PinotTableCreationTool;
+import org.hypertrace.core.serviceframework.PlatformService;
+import org.hypertrace.core.serviceframework.config.ConfigClient;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/**
- * ViewCreatorLauncher creates Pinot tables as specified in the config.
- */
-public class ViewCreatorLauncher {
+/** ViewCreatorLauncher creates Pinot tables as specified in the config. */
+public class ViewCreatorLauncher extends PlatformService {
   private static final Logger LOGGER = LoggerFactory.getLogger(ViewCreatorLauncher.class);
+  private static final String TOOL_CLASS = "tool.class";
+  private static final String VIEWS = "views";
+  private static final String SERVICE_NAME_CONFIG = "service.name";
 
-  public static void main(String[] args) {
-    updateRuntime();
-    launch(args);
+  public ViewCreatorLauncher(ConfigClient configClient) {
+    super(configClient);
   }
 
-  private static void launch(String[] args) {
-    String configFilePath = null;
-    if (args.length > 0) {
-      configFilePath = args[0];
-      LOGGER.info("Loading config file path: {}", configFilePath);
-    } else {
-      LOGGER.error("A config file path is required to launch the job/services.");
-      System.exit(1);
-    }
-    Config fileConfig = ConfigFactory.parseFile(new File(configFilePath));
-    Config configs = ConfigFactory.load(fileConfig);
+  @Override
+  protected void doInit() {}
 
-    final String mainClass = configs.getString("mainClass");
-    switch (mainClass) {
-      case "org.hypertrace.core.viewcreator.pinot.PinotTableCreationTool":
-        try {
-          ViewCreationSpec viewCreationSpec = ViewCreationSpec
-              .parse(configs);
-          PinotTableCreationTool tool = new PinotTableCreationTool(viewCreationSpec);
-          tool.execute();
-        } catch (Exception e) {
-          LOGGER.error("Exception while executing PinotTableCreationTool.execute()", e);
-          System.exit(1);
-        }
-        break;
-      case "org.hypertrace.core.viewcreator.pinot.PinotMultiTableCreationTool":
-        try {
-          PinotMultiTableCreationTool tool = new PinotMultiTableCreationTool(configs.getStringList("views"));
-          tool.execute();
-        } catch (Exception e) {
-          LOGGER.error("Exception while executing PinotMultiTableCreationTool.execute()", e);
-          System.exit(1);
-        }
-        break;
-      default:
-        System.exit(1);
-        LOGGER.error("Cannot find the class to execute. Did not create the tables");
+  @Override
+  protected void doStart() {
+    try {
+      getTableCreationTool(getAppConfig()).create();
+      shutdown();
```

Comment:
I'm not sure if you need to explicitly call shutdown(). What happens when you remove this?

Reply:
view creator job stays in running state and installation can't proceed.

Reply:
oh ok then.